.gitlab-ci.yml

```diff
@@ -1,118 +1,142 @@
+image: stackstate/stackstate-agent-runner-circle:omnibus-cache
+
 stages:
-- tests
-- build
+- prepare
+- build_and_test
 - publish
 
-unit_tests:
- stage: tests
- image: stackstate/stackstate-agent-runner-circle:latest
- before_script:
+before_script:
  - mkdir -p /go/src/github.com/StackVista
  - ln -s $CI_PROJECT_DIR  /go/src/github.com/StackVista/stackstate-agent
- script:
+   # Either we already have a venv, or we still need to make it (as part of get_deps.sh)
+ - source $CI_PROJECT_DIR/venv/bin/activate || source .gitlab-scripts/ensure_deps.sh
  - cd /go/src/github.com/StackVista/stackstate-agent
- - pip install -r requirements.txt
- - inv deps
- - inv -e agent.build --race --precompile-only
- - inv -e test --coverage --race --profile --fail-on-fmt --cpus 4 --build-exclude clusterchecks,kubeapiserver,kubelet
+
+dependencies_clear_cache:
```

Comment:
is this `dependencies_clear_cache` used anywhere ?

Reply:
yes, it is a manual job that we can use to clear the cache